common/js/src/logging/logging.js

```diff
@@ -195,14 +194,13 @@ GSC.Logging.checkWithLogger = function(
  * @param {...*} var_args The items to substitute into the failure message.
  */
 GSC.Logging.fail = function(opt_message, var_args) {
-  var messageAndArgs = Array.prototype.slice.call(arguments);
   if (goog.DEBUG) {
-    goog.asserts.fail.apply(goog.asserts, messageAndArgs);
+    throwAssertionError(opt_message, var_args);
   } else {
+    var messageAndArgs = Array.prototype.slice.call(arguments);
```

Comment:
This is a trick to transform arguments variable (which is for some reason not an array) into an array. I believe it's a pretty common way of doing this, e.g. see the sample code in the Array.slice docs in MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#Array-like_objects